src/internal/connector/graph_connector_onedrive_test.go

```diff
@@ -110,6 +120,26 @@ func (suite *GraphConnectorOneDriveIntegrationSuite) SetupSuite() {
 	suite.secondaryUser = tester.SecondaryM365UserID(suite.T())
 	suite.acct = tester.NewM365Account(suite.T())
 
+	// Not using m365.Users to avoid cyclic imports
+	users, err := discovery.Users(ctx, suite.connector.Owners.Users(), fault.New(true))
+	require.NoError(suite.T(), err, "fetching users")
+
+	for _, u := range users {
+		if *u.GetUserPrincipalName() == suite.user {
+			suite.userID = *u.GetId()
+		} else if *u.GetUserPrincipalName() == suite.secondaryUser {
+			suite.secondaryUserID = *u.GetId()
+		}
+	}
+
+	if suite.userID == "" {
+		require.FailNowf(suite.T(), "unable to find user id", "user %d", suite.user)
+	}
+
+	if suite.secondaryUserID == "" {
+		require.FailNowf(suite.T(), "unable to find user id", "user %d", suite.secondaryUser)
+	}
```

Comment:
nit: you can use `require.NotEmpty()` for these checks
```suggestion
	require.NotEmptyf(suite.T(), suite.userID, "unable to find user id", "user %d", suite.user)
	}

	require.NoteEmptyf(suite.T(), suite.secondaryUserID, "unable to find user id", "user %d", suite.secondaryUser)
```